src/process-events.c

```diff
@@ -616,34 +610,30 @@ static __always_inline int exit_exec(struct pt_regs *__ctx, u32 i_dev, u64 i_ino
     if (!id)
         goto Flush;
 
-    ptelemetry_event_t ev = &(telemetry_event_t){
-        .id = 0,
-        .done = FALSE,
-        .telemetry_type = TE_UNSPEC,
-        .u.v = {
-            .value[0] = '\0',
-            .truncated = FALSE,
-        },
-    };
-    ev->id = *id;
     bpf_map_delete_elem(&process_ids, &pid_tgid);
 
-    file_info_t fi = {
-        .inode = i_ino,
-        .devmajor = MAJOR(i_dev),
-        .devminor = MINOR(i_dev),
-    };
+    u32 retcode = (u32)PT_REGS_RC(__ctx);
+    if (retcode != 0) {
+        goto Flush;
+    }
```

Comment:
Tested with the below program and can confirm that an event is emit for `ls` but not for `lol`.

```
#include <unistd.h>

void main() {
	char *arg[] = { "/no/file/lol", NULL };
	char *env[] = { NULL };
	execve("/no/file/lol", arg, env);
	execve("/usr/bin/ls", arg, env);
}
```